src/sentry/tasks/derive_code_mappings.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from datetime import timedelta
+from typing import Any, List, Mapping, Optional, Set, Tuple
+
+from django.utils import timezone
+
+from sentry.db.models.fields.node import NodeData
+from sentry.models import Project
+from sentry.models.group import Group
+from sentry.models.organization import Organization, OrganizationStatus
+from sentry.tasks.base import instrumented_task
+from sentry.utils.safe import get_path
+
+ACTIVE_PROJECT_THRESHOLD = timedelta(days=7)
+GROUP_ANALYSIS_RANGE = timedelta(days=14)
+
+logger = logging.getLogger("sentry.tasks.derive_code_mappings")
+
+
+@instrumented_task(  # type: ignore
+    name="sentry.tasks.derive_code_mappings.identify_stacktrace_paths",
+    queue="derive_code_mappings",
+    max_retries=0,  # if we don't backfill it this time, we'll get it the next time
+)
+def identify_stacktrace_paths(
+    organizations: Optional[List[Organization]] = None,
+) -> Mapping[str, Mapping[str, List[str]]]:
+    """
+    Generate a map of projects to stacktrace paths for specified organizations,
+    or all active organizations if unspecified.
+
+    This filters out projects have not had an event in the last 7 days or have
+    non-python files in the stacktrace.
+    """
+    if organizations is None:
+        organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+
+    filename_maps = {}
+    for org in organizations:
+        projects = Project.objects.filter(organization=org, first_event__isnull=False)
+        projects = [
+            project
+            for project in projects
+            if Group.objects.filter(
+                project=project,
+                last_seen__gte=timezone.now() - ACTIVE_PROJECT_THRESHOLD,
+            ).exists()
+        ]
+
+        project_file_map = {project.slug: get_all_stacktrace_paths(project) for project in projects}
+        filename_maps[org.slug] = project_file_map
+    return filename_maps
+
+
+def get_all_stacktrace_paths(project: Project) -> List[str]:
+    groups = Group.objects.filter(
+        project=project, last_seen__gte=timezone.now() - GROUP_ANALYSIS_RANGE
+    )
+
```

Comment:
This could be a significant number of groups, depending on project. If it is too many, you might hit OOM issues.

Since you're just processing one at a time, you could use `RangeQuerysetWrapper` to keep memory usage bounded.

I'm also not totally sure you need to fetch all groups from this time range. It seems like you mostly just want to sample a few groups and check their stack trace? Another way to do this is to query snuba for events from the last `GROUP_ANALYSIS_RANGE` and get a distinct list of platforms from them. That would allow this to be done with one query per project

Reply:
Actually, even better, you can make use of `ProjectPlatform`, which is generated by this task https://github.com/getsentry/sentry/blob/fe07466a1449a5ae60526528ce7bf9399b59b47d/tests/acceptance/test_emails.py#L63-L62

Reply:
Actually I think I'm misunderstanding what this is doing, where are these paths being used in general?